projects/js-packages/charts/src/components/line-chart/line-chart.tsx

```diff
@@ -67,6 +66,35 @@ const renderTooltip = ( {
 	);
 };
 
+// Validation functions
+const isValidDate = ( date: Date ) => ! isNaN( date.getTime() );
+const isValidValue = ( value: number | null | undefined ): value is number =>
+	typeof value === 'number' && ! isNaN( value );
+
+const validateData = ( data: SeriesData[] ) => {
+	if ( ! data?.length ) {
+		return 'no data available';
+	}
+
+	for ( const series of data ) {
+		if ( ! series.data?.length ) {
+			return 'invalid data';
+		}
+
+		for ( const point of series.data ) {
+			if (
+				! ( 'date' in point ) ||
+				! isValidDate( point.date as Date ) ||
```

Comment:
I understand we only allow date types as X markers, but we probably shouldn't because they takes away the ability the format the date / text. Not related to the PR tho.

Reply:
Good point. Let's tackle that in another PR 👍